Sorting/quick_sort/quick_sort.cpp.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+QUICK SORT ALGORITHM : 	TIME COMPLEXITY O(nlogn(n))
+
+*/
+
+
+#include<iostream>
+using namespace std;
+
+//note: call by address in swap function 
+void swap(int *a,int *b)
+{
+	int t;
+	t=*a;
+	*a=*b;
+	*b=t;
+}
+
+
+//partition function which makes partition in array and return the partition index
+int partition(int a[],int low ,int high)
+{
+	// i is leftmost index and j is the rightmost index
+	int i = low,j = high;     
+	int pivot = a[low];     
+	
+	
+	do{
+		do
+		{
+		  i++;		
+		}while(a[i]<=pivot);  //note :condition is for continuation of loop ans not for stop
+		
+		do
+		{
+			j--;
+		}while(a[j]>pivot);   // note: condition is for continuation of loop and not for stop
+		
+		
+		if(i<j)              // note: condition
+			swap(&a[i],&a[j]);  // note:
+			
+	}while(i<j);              //note:condition
+	
+	swap(&a[low],&a[j]); //note:
+    
+	return j;  
+}
+
+
+//Quick sort function
+void Quick_sort(int a[],int low,int high)
+{
+	int j; // j is partition index
+	
+	 //note: This condition for says that min two elements are required for quick sort
+	if(low < high)          
+	{
+		//index j is partition index 
+		j = partition(a,low,high) ;  
+		
+		// left partition  // note: low to j is for left partition
+		Quick_sort(a,low,j);       
+		
+		//right partition    //note: j+1 to high is for right partition
+		Quick_sort(a,j+1,high);  
+		
+	}
+}
+
+
+//main function
+int main(){
+	
+	int arr[]= {2,4,3,6,9,8, -1};	
+	int n = sizeof(arr)/sizeof(arr[0]);
+		
```

Comment:
Can we take this input from STDIN.
First line: N - number of inputs
Second line: N space-separated integers

for example, 
```
5
2 5 23 14 4
```